src/transactions/transaction.rs

```diff
@@ -0,0 +1,881 @@
+    ///
+    /// [`Busy`]: crate::ErrorKind::Busy
+    /// [`TimedOut`]: crate::ErrorKind::TimedOut
+    /// [`TryAgain`]: crate::ErrorKind::TryAgain
+    /// [`MergeInProgress`]: crate::ErrorKind::MergeInProgress
+    /// [`TransactionDB`]: crate::TransactionDB
+    pub fn merge_cf<K: AsRef<[u8]>, V: AsRef<[u8]>>(
+        &self,
+        cf: &impl AsColumnFamilyRef,
+        key: K,
+        value: V,
+    ) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_merge_cf(
+                self.inner,
+                cf.inner(),
+                key.as_ref().as_ptr() as *const c_char,
+                key.as_ref().len() as size_t,
+                value.as_ref().as_ptr() as *const c_char,
+                value.as_ref().len() as size_t
+            ));
+            Ok(())
+        }
+    }
+
+    /// Delete the key value if it exists and do conflict checking on the key.
+    ///
+    /// See [`delete_cf`] for details.
+    ///
+    /// [`delete_cf`]: Self::delete_cf
+    pub fn delete<K: AsRef<[u8]>>(&self, key: K) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_delete(
+                self.inner,
+                key.as_ref().as_ptr() as *const c_char,
+                key.as_ref().len() as size_t
+            ));
+        }
+        Ok(())
+    }
+
+    /// Delete the key value in the given column family and do conflict checking.
+    ///
+    /// If this transaction was created by a [`TransactionDB`], it can return error of kind:
+    /// * [`Busy`] if there is a write conflict.
+    /// * [`TimedOut`] if a lock could not be acquired.
+    /// * [`TryAgain`] if the memtable history size is not large enough.
+    /// * [`MergeInProgress`] if merge operations cannot be resolved.
+    /// * or other errors on unexpected failures.
+    ///
+    /// [`Busy`]: crate::ErrorKind::Busy
+    /// [`TimedOut`]: crate::ErrorKind::TimedOut
+    /// [`TryAgain`]: crate::ErrorKind::TryAgain
+    /// [`MergeInProgress`]: crate::ErrorKind::MergeInProgress
+    /// [`TransactionDB`]: crate::TransactionDB
+    pub fn delete_cf<K: AsRef<[u8]>>(
+        &self,
+        cf: &impl AsColumnFamilyRef,
+        key: K,
+    ) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_delete_cf(
+                self.inner,
+                cf.inner(),
+                key.as_ref().as_ptr() as *const c_char,
+                key.as_ref().len() as size_t
+            ));
+        }
+        Ok(())
+    }
+
+    pub fn iterator<'a: 'b, 'b>(
+        &'a self,
+        mode: IteratorMode,
+    ) -> DBIteratorWithThreadMode<'b, Self> {
+        let readopts = ReadOptions::default();
+        self.iterator_opt(mode, readopts)
+    }
+
+    pub fn iterator_opt<'a: 'b, 'b>(
+        &'a self,
+        mode: IteratorMode,
+        readopts: ReadOptions,
+    ) -> DBIteratorWithThreadMode<'b, Self> {
+        DBIteratorWithThreadMode::new(self, readopts, mode)
```

Comment:
It's wrong. [Transaction](https://github.com/facebook/rocksdb/blob/main/include/rocksdb/c.h#L2579) and [TransactionDB](https://github.com/facebook/rocksdb/blob/main/include/rocksdb/c.h#L2588) have its own implementations for returning iterators.

Reply:
Transaction also implements `DBAccess` and call [rocksdb_transaction_create_iterator](https://github.com/yiyuanliu/rust-rocksdb/blob/transaction/src/transactions/transaction.rs#L48) in `create_iterator`.